pkg/kexec/kexec_linux_amd64.go

```diff
@@ -24,8 +27,43 @@ func FileLoad(kernel, ramfs *os.File, cmdline string) error {
 		flags |= unix.KEXEC_FILE_NO_INITRAMFS
 	}
 
+	if rspd := acpiRSDP(); rspd != "" {
+		cmdline += " acpi_rsdp="+rspd
+	}
+
 	if err := unix.KexecFileLoad(int(kernel.Fd()), ramfsfd, cmdline, flags); err != nil {
 		return fmt.Errorf("sys_kexec(%d, %d, %s, %x) = %v", kernel.Fd(), ramfsfd, cmdline, flags, err)
 	}
 	return nil
 }
+
+func acpiRSDP() string {
```

Comment:
This is from cmds/rsdp.go

Now that we have two things looking for RSDP, my suggestion is to put the cmds/rsdp.go function in pkg/uroot/util and use that function in both packages. This would also give us a chance to pick a consistent place to get the rsdp.

but we definitely don't want two commands with two different ways to get rsdp, so adding a function to pkg/uroot/util makes the most sense.


func getRSDP(path string) (string, error) {
	fd, err := unix.Open(path, unix.O_RDONLY|unix.O_NONBLOCK, 0)
	if err != nil {
		log.Fatal(err)
	}
	file := os.NewFile(uintptr(fd), "kernel messages")
	defer file.Close()

	s := bufio.NewScanner(file)
	for s.Scan() {
		if err := s.Err(); err != nil {
			return "", err
		}
		res := s.Text()
		if strings.Contains(res, "RSDP") {
			rv := strings.Split(res, " ")
			if len(res) < 3 {
				continue
			}
			return rv[2], nil
		}
	}
	return "", fmt.Errorf("Could not find RSDP")
}

Reply:
I do not think we need two implementations of this, we should get rid of  cmds/rsdp.go. If you need to get the RDSP on the command line you can just do:
# sed -n -e 's/^ACPI20=//p' /sys/firmware/efi/systab
0x...

That command is most likely needed to call kexec with its output. With this commit it won't be needed anyways.

Reply:
you don't have sed so that won't work :-)

Reply:
Well, in any case, is cmds/rsdp.go used for anything else besides calling kexec?